src/client.ts

```diff
@@ -4835,30 +4933,37 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
             $txnId: txnId,
         };
 
+        const futurePathPart = futureOpts ? "_future" : "";
         let path: string;
 
         if (event.isState()) {
-            let pathTemplate = "/rooms/$roomId/state/$eventType";
+            let pathTemplate = `/rooms/$roomId/state${futurePathPart}/$eventType`;
             if (event.getStateKey() && event.getStateKey()!.length > 0) {
-                pathTemplate = "/rooms/$roomId/state/$eventType/$stateKey";
+                pathTemplate = `/rooms/$roomId/state${futurePathPart}/$eventType/$stateKey`;
             }
             path = utils.encodeUri(pathTemplate, pathParams);
         } else if (event.isRedaction() && event.event.redacts) {
-            const pathTemplate = `/rooms/$roomId/redact/$redactsEventId/$txnId`;
+            // TODO: support future redactions?
+            const pathTemplate = `/rooms/$roomId/redact${futurePathPart}/$redactsEventId/$txnId`;
```

Comment:
Is this actually what we want/need.
If i understand correctly this is to send a redact that does not happen now but only once the future resolves?
I think we can simply use the send endpoint and choose the event type and the content accordingly to send a redact event as a future.
Spcifying a custom future_readact endpoint maybe should be a seperate MSC since it only would be "alternative syntax"

Reply:
The `/redact/` endpoint has different semantics than `/send/` (eg. it targets a specific eventID, provides a reason for the redaction, and abstracts what data is even removed from the event), so AIUI it's different enough to warrant another future-compatible endpoint.  Though I don't think we need it right now so I'm satisfied with just throwing a "not implemented" error here.

Reply:
O just played around and tried sending a custom timeline events and it seems it does not work right now. there was a thread somewhere discussing this and iirc it was not far from making this work. That would help that we done need a custom redact future endpoint but instead do sth like this:
![image](https://github.com/user-attachments/assets/fff038b4-ca21-44bc-8b9d-8aba3ad57528)